src/theory/strings/theory_strings.cpp

```diff
@@ -604,6 +604,28 @@ TrustNode TheoryStrings::expandDefinition(Node node)
     return TrustNode::mkTrustRewrite(node, ret, nullptr);
   }
 
+  if (node.getKind() == SEQ_NTH)
+  {
+    // str.nth(s,i) --->
+    //   witness k. ite(0 <= i < len(s), unit(k) = seq.at(s,i), true)
```

Comment:
Isn't this just:
```
witness k. 0 <= i < len(s) => unit(k) = seq.at(s,i)
```?

Reply:
Actually, I think that this needs to be:
```
witness k. ite(0 <= i < len(s), unit(k) = seq.at(s,i), USeq(s,i))
```

where `USeq(s,i)` is an uninterpreted function.

Reply:
Ah, I guess Andy's test case would catch this as well :)

Reply:
Yes, that was the point of my test :)  And yes, we need the uninterpreted function.